src/app/consumption-estimation/consumption-estimation.component.ts

```diff
@@ -0,0 +1,206 @@
+    });
+    this.consumptionDetailsForm = this.formBuilder.group({
+        yearlyConsumption: ['',Validators.required],
+        selectedEquipments : new FormArray([])
+    });
+  }
+  get service() { return this.serviceChoiceForm.controls; }
+  get familyAddress() { return this.familyAdressCompoundForm.controls; }
+  get meterType() {return this.meterTypeForm.controls;}
+  get consumption() { return this.consumptionDetailsForm.controls; }
+
+  onCheckboxChange(event: any) {
+    const selectedEquipments = (this.consumption['selectedEquipments'] as FormArray);
+    if (event.target.checked) {
+      selectedEquipments.push(new FormControl(event.target.value));
+    } else {
+      const index = selectedEquipments.controls.findIndex(x => x.value === event.target.value);
+      selectedEquipments.removeAt(index);
+    }
+  }
+
+
+next()
+{
+  if(this.step==1)
+  {
+    this.service_step=true;
+    if(this.serviceChoiceForm.invalid){return}
+    this.step++;
+    console.log(this.step);
+  }
+  if(this.step==2)
+  {
+    console.log(this.step);
+    this.familyAdress_step=true;
+    if(this.familyAdressCompoundForm.invalid)
+    {
+      console.error();
+      return
+    }
+    this.step++;
+    console.log(this.step);
+  }
+  if(this.step==3)
+  {
+    console.log(this.step);
+    this.meterType_step=true;
+    if(this.meterTypeForm.invalid)
+    {
+      console.error();
+      return
+    }
+    this.step++;
+    console.log(this.step);
+  }
+}
+previous()
+{
+  this.step--;
+  console.log(this.step);
+  if(this.step==1)
+  {
+    this.service_step=false;
+    console.log(this.step);
+  }
+  if(this.step==2)
+  {
+    this.familyAdress_step=false;
+    console.log(this.step);
+  }
+  if(this.step==3)
+  {
+    this.meterType_step=false;
+    console.log(this.step);
+  }
+  
+}
+
+calculateConsumption()
+{
+  let estimation:number=0;
+  let nrOfPeople : number = this.familyAdressCompoundForm.get('members')?.value;
+  console.log("ppl",nrOfPeople);
+  if(nrOfPeople==1){
+      estimation=18;
+  }
+  else if(nrOfPeople==2)
+  {
+    estimation=28;
+  }
+  else if(nrOfPeople==3) 
+  { 
+    estimation=40;
+  }
+  else
+  {
+  estimation=60;}
+     
+  console.log("est",estimation);
+  
+  let building : number = this.familyAdressCompoundForm.get('houseType')?.value;
+  console.log("bTy",building);
+  if(building==1){
+    estimation-=5;
+  }
+  else if(building==2)
+  {
+    estimation-=2;
+  }
+  else if(building==3) 
+  { 
+    estimation+=5;
+  }
+
+  console.log("est",estimation);
+
+  let appliances: [number]=(this.consumptionDetailsForm.get('selectedEquipments')?.value);
+  console.log(appliances);
+  appliances.forEach(app => {
+    if(app==1)
+    {estimation+=1}
+    if(app==2)
+    {estimation+=0.20}
+    if(app==3)
+    {estimation+=0.36}
+    if(app==4)
+    {estimation+=0.30}
+    if(app==5)
+    {estimation+=1.07}
+  });
+  console.log(estimation);
+
+  
+  return estimation;
+}
+
+submit()
+{
+  if(this.step==4)
```

Comment:
It is  momentarily like that for testing purposes. Because its main function is to send the form's information to employees so they can validate, which I'm working on.

Reply:
Not sure what u mean. Just meant that the button to call submit would only be in the last step, so it's a redundant function I think.